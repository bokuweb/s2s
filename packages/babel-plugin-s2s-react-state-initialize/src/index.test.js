// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const getFixturePath = (...x) =>
  path.resolve(__dirname, '..', '__fixtures__', ...x)

function testWithFixture(title, fixture) {
  pluginTester({
    plugin,
    snapshot: true,
    pluginOptions: { from: fixture },
    tests: [
      {
        title,
        fixture,
      },
    ],
  })
}

testWithFixture('initialize react state', getFixturePath('basic', 'index.js'))
testWithFixture('no react class', getFixturePath('no-react', 'index.js'))
testWithFixture('no state type', getFixturePath('no-state', 'index.js'))
