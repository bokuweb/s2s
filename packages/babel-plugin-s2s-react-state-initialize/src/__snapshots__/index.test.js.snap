// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`s2s-react-state-initialize initialize react state: initialize react state 1`] = `
"
// @flow
import * as React from 'react'

type C = { d: string, e: number }

type State = { a: string, b: number, c: C }

// $FlowFixMe
class Fixture extends React.Component<{}, State> {
  state = {}
  render() {
    return null
  }
}

export default Fixture

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import * as React from 'react';

type C = { d: string; e: number; };

type State = { a: string; b: number; c: C; };

// $FlowFixMe
class Fixture extends React.Component<{}, State> {
  state = {};
  render() {
    return null;
  }
}

export default Fixture;
"
`;

exports[`s2s-react-state-initialize no react class: no react class 1`] = `
"
// @flow
import * as React from 'react'

class Fixture {
  render() {
    return null
  }
}

export default Fixture

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import * as React from 'react';

class Fixture {
  render() {
    return null;
  }
}

export default Fixture;
"
`;

exports[`s2s-react-state-initialize no state type: no state type 1`] = `
"
// @flow
import * as React from 'react'

type C = { d: string, e: number }

type State = { a: string, b: number, c: C }

class Fixture extends React.Component<{}> {
  render() {
    return null
  }
}

export default Fixture

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import * as React from 'react';

type C = { d: string; e: number; };

type State = { a: string; b: number; c: C; };

class Fixture extends React.Component<{}> {
  render() {
    return null;
  }
}

export default Fixture;
"
`;
